pandas/core/indexes/base.py

```diff
@@ -3864,7 +3864,21 @@ def dropna(self, how='any'):
         return self._shallow_copy()
 
     def _evaluate_with_timedelta_like(self, other, op, opstr, reversed=False):
-        raise TypeError("can only perform ops with timedelta like values")
+        # Timedelta knows how to operate with np.array, so dispatch to that
```

Comment:
shouldn't this be left alone, and rather define this in ``TimedeltaIndex``?

Reply:
This for for numeric-dtyped indexes.  e.g. `pd.Index(range(3)) * pd.Timedelta(days=1)` goes through this path.

Reply:
why doesn't this just raise NotImplementedError? which then Timedelta would handle? or is that too recursive a path?

Reply:
For one thing b/c there’s pytimedelta and timedelta64 that’s need to be handled.